src/components/Form/FormField.jsx

```diff
@@ -1,83 +1,114 @@
 import React from 'react';
-import { connect } from 'react-redux';
+import {
+  connect,
+} from 'react-redux';
 import AutoComplete from 'material-ui/AutoComplete';
-import { addChip, setFieldText } from 'actions';
-import { form } from 'reducers';
+import {
+  addChip,
+} from 'actions';
+import {
+  form,
+} from 'reducers';
 import getClosestMatch from './utility';
 import ChipList from './ChipList';
 import styles from '../palette.css';
 
-const onNewRequest = ({ value, index, formName, name, dataSourceConfig, dataSource, strict, addChip, setFieldText, limit }) => {
-  if (index !== -1) {
-    addChip(
-      formName,
-      name, {
-        text: value[dataSourceConfig.text],
-        value: value[dataSourceConfig.value],
-      },
-      limit
-    );
-    setFieldText(formName, name, '');
-  }
-  if (strict && index === -1) {
-    const closestMatch = getClosestMatch(
-      dataSource, dataSourceConfig,
-      AutoComplete.fuzzyFilter, value[dataSourceConfig.text] || value
-    );
-    // We can't just do truthy check here because the value could be 0
-    if (closestMatch.value || closestMatch.value === 0) {
+class FormField extends React.Component {
+  handleRequest({
+    value,
+    index,
+    formName,
+    name,
+    dataSourceConfig,
+    dataSource,
+    strict,
+    addChip,
+    limit,
+  }) {
+    if (index !== -1) {
+      addChip(
+        formName,
+        name, {
+          text: value[dataSourceConfig.text],
+          value: value[dataSourceConfig.value],
+        },
+        limit
+      );
+    }
+    if (strict && index === -1) {
+      const closestMatch = getClosestMatch(
+        dataSource, dataSourceConfig,
+        AutoComplete.fuzzyFilter, value[dataSourceConfig.text] || value
+      );
+      // We can't just do truthy check here because the value could be 0
+      if (closestMatch.value || closestMatch.value === 0) {
+        addChip(
+          formName,
+          name, {
+            text: closestMatch.text,
+            value: closestMatch.value,
+          },
+          limit
+        );
+      }
+    }
+    if (!strict && index === -1) {
       addChip(
         formName,
         name, {
-          text: closestMatch.text,
-          value: closestMatch.value,
+          text: value,
+          value,
         },
         limit
       );
-      setFieldText(formName, name, '');
     }
+    this.input.setState({
```

Comment:
You might want to put a comment on this (not doing it here causes the thing not to clear, right?)